catchup/service_test.go

```diff
@@ -181,6 +181,100 @@ func (cl *periodicSyncLogger) Warnf(s string, args ...interface{}) {
 	cl.Logger.Warnf(s, args...)
 }
 
+func TestSyncRound(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	// Make Ledger
+	local := new(mockedLedger)
+	local.blocks = append(local.blocks, bookkeeping.Block{})
+
+	remote, _, blk, err := buildTestLedger(t, bookkeeping.Block{})
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	addBlocks(t, remote, blk, 10)
+
+	// Create a network and block service
+	blockServiceConfig := config.GetDefaultLocal()
+	net := &httpTestPeerSource{}
+	ls := rpcs.MakeBlockService(logging.Base(), blockServiceConfig, remote, net, "test genesisID")
+
+	nodeA := basicRPCNode{}
+	nodeA.RegisterHTTPHandler(rpcs.BlockServiceBlockPath, ls)
+	nodeA.start()
+	defer nodeA.stop()
+	rootURL := nodeA.rootURL()
+	net.addPeer(rootURL)
+
+	auth := &mockedAuthenticator{fail: true}
+	initialLocalRound := local.LastRound()
+	require.True(t, 0 == initialLocalRound)
+
+	// Make Service
+	localCfg := config.GetDefaultLocal()
+	// Set MaxAcctLookback to 1 just so we have a small number of rounds between sync round and latest
+	localCfg.MaxAcctLookback = 1
+	s := MakeService(logging.Base(), localCfg, net, local, auth, nil, nil)
+	s.log = &periodicSyncLogger{Logger: logging.Base()}
+	s.deadlineTimeout = 2 * time.Second
+
+	// Set sync round success
+	err = s.SetSyncRound(uint64(initialLocalRound + 2))
+	require.NoError(t, err)
+
+	s.Start()
+	defer s.Stop()
+	// wait past the initial sync - which is known to fail due to the above "auth"
+	time.Sleep(s.deadlineTimeout*2 - 200*time.Millisecond)
+	require.Equal(t, initialLocalRound, local.LastRound())
+	auth.alter(-1, false)
+
+	// wait until the catchup is done. Since we've might have missed the sleep window, we need to wait
+	// until the synchronization is complete.
+	waitStart := time.Now()
+	for time.Now().Sub(waitStart) < 2*s.deadlineTimeout {
+		if remote.LastRound() == local.LastRound() {
+			break
+		}
+		time.Sleep(20 * time.Millisecond)
+	}
+	// Assert that the last block is the one we expect--i.e. syncRound(inclusive)..syncRound+MaxAccountLookback(non-inclusive)
+	_, rnd, err := s.GetSyncRound()
+	require.NoError(t, err)
+	rr, lr := basics.Round(rnd+s.cfg.MaxAcctLookback-1), local.LastRound()
+	require.Equal(t, rr, lr)
+
+	for r := basics.Round(1); r < rr; r++ {
+		localBlock, err := local.Block(r)
+		require.NoError(t, err)
+		remoteBlock, err := remote.Block(r)
+		require.NoError(t, err)
+		require.Equal(t, remoteBlock.Hash(), localBlock.Hash())
+	}
+
+	// unset syncRound and make sure we finish catching up
+	err = s.UnsetSyncRound()
+	require.NoError(t, err)
+	// wait until the catchup is done
+	waitStart = time.Now()
+	for time.Now().Sub(waitStart) < 8*s.deadlineTimeout {
+		if remote.LastRound() == local.LastRound() {
+			break
+		}
+		time.Sleep(20 * time.Millisecond)
+	}
+	rr, lr = remote.LastRound(), local.LastRound()
+	require.Equal(t, rr, lr)
+	for r := basics.Round(1); r < remote.LastRound(); r++ {
```

Comment:
nit:
```suggestion
	for r := basics.Round(1); r < rr; r++ {
```
and should it be r<=rr?